Box.V2/Managers/BoxFilesManager.cs

```diff
@@ -374,8 +374,9 @@ public BoxFilesManager(IBoxConfig config, IBoxService service, IBoxConverter con
         /// <param name="fileName">Name of the remote file name.</param>
         /// <param name="fileSize">Size of the file.</param>
         /// <param name="folderId">parent folder id.</param>
+        /// <param name="sha1HashFunction">Function delegate that returns message digest of the stream, formatted as specified by RFC 3230.</param>
         /// <returns></returns>
-        public async Task<bool> UploadUsingSessionAsync(Stream stream, string fileName, long fileSize, string folderId)
+        public async Task<bool> UploadUsingSessionAsync(Stream stream, string fileName, long fileSize, string folderId, Func<Stream,string> sha1HashFunction)
```

Comment:
why sha1HashFunction is on the public interface, could we hide it and auto select internally?